services/motion/explore/explore.go

```diff
@@ -0,0 +1,571 @@
+	component, ok := ms.components[baseName]
+	if !ok {
+		return nil, resource.DependencyNotFoundError(baseName)
+	}
+
+	b, ok := component.(base.Base)
+	if !ok {
+		return nil, fmt.Errorf("cannot get component of type %T because it is not a Case", component)
+	}
+
+	// Generate kinematic base options from motionCfg
+	kinematicsOptions := kinematicbase.NewKinematicBaseOptions()
+	kinematicsOptions.NoSkidSteer = true
+	kinematicsOptions.UsePTGs = false
+	kinematicsOptions.PositionOnlyMode = true
+	kinematicsOptions.AngularVelocityDegsPerSec = motionCfg.AngularDegsPerSec
+	kinematicsOptions.LinearVelocityMMPerSec = motionCfg.AngularDegsPerSec
+
+	// Create new kinematic base (differential drive)
+	kb, err := kinematicbase.WrapWithKinematics(
+		ctx,
+		b,
+		ms.logger,
+		nil,
+		[]referenceframe.Limit{{Min: -moveLimit, Max: moveLimit}, {Min: -moveLimit, Max: moveLimit}},
+		kinematicsOptions,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return kb, nil
+}
+
+// createObstacleDetectors will generate the list of obstacle detectors from the camera and vision services
+// names provided in them motionCfg.
+func (ms *explore) createObstacleDetectors(motionCfg motion.MotionConfiguration) ([]obstacleDetectorObject, error) {
+	var obstacleDetectors []obstacleDetectorObject
+
+	// Iterate through obstacleDetectorsNames
+	for _, obstacleDetectorsName := range motionCfg.ObstacleDetectors {
+		// Select the vision service from the service list using the vision service name in obstacleDetectorsNames
+		visionServiceResource, ok := ms.services[obstacleDetectorsName.VisionServiceName]
+		if !ok {
+			return nil, resource.DependencyNotFoundError(obstacleDetectorsName.VisionServiceName)
+		}
+		visionService, ok := visionServiceResource.(vision.Service)
+		if !ok {
+			return nil, fmt.Errorf("cannot get service of type %T because it is not a vision service",
+				visionServiceResource,
+			)
+		}
+
+		// Select the camera from the component list using the camera name in obstacleDetectorsNames
+		// Note: May need to be converted to a forloop if we accept multiple cameras for each vision service
+		cameraResource, ok := ms.components[obstacleDetectorsName.CameraName]
+		if !ok {
+			return nil, resource.DependencyNotFoundError(obstacleDetectorsName.CameraName)
+		}
+		cam, ok := cameraResource.(camera.Camera)
+		if !ok {
+			return nil, fmt.Errorf("cannot get component of type %T because it is not a camera", cameraResource)
+		}
+		obstacleDetectors = append(obstacleDetectors, obstacleDetectorObject{visionService: cam})
+	}
+
+	return obstacleDetectors, nil
+}
+
+// createMotionPlan will construct a motion plan using the given destination TBD.
+func (ms *explore) createMotionPlan(
+	ctx context.Context,
+	kb kinematicbase.KinematicBase,
+	destination *referenceframe.PoseInFrame,
+	worldState *referenceframe.WorldState,
+	extra map[string]interface{},
+) (motionplan.Plan, error) {
+	fs, err := ms.fsService.FrameSystem(ctx, worldState.Transforms())
```

Comment:
Simplification added, I have also removed worldState as an input to this function. One thing to confirm on this: we create a new worldState both in `createMotionPlan` and in ever iteration of the checkForObstacles function. However, we do not send the initial worldState from `createMotionPlan` to `checkForObstacles` because `checkForObstacles` is dealing with the transient worldState. Does this make sense?